push_notifications/gcm.py

```diff
@@ -47,11 +52,11 @@ def _gcm_send(data, content_type):
 		"Content-Length": str(len(data)),
 	}
 	request = Request(SETTINGS["GCM_POST_URL"], data, headers)
-	return urlopen(request, timeout=SETTINGS["GCM_ERROR_TIMEOUT"]).read().decode("utf-8")
+	return decodestr(urlopen(request, timeout=SETTINGS["GCM_ERROR_TIMEOUT"]).read())
 
 
-def _fcm_send(data, content_type):
-	key = SETTINGS.get("FCM_API_KEY")
+def _fcm_send(data, content_type, application_id):
+	key = get_gcm_api_key(application_id)
```

Comment:
The code means that either GCM, or FCM API KEY is used for the application, not the both simultaneously.

I also really wanted to have some proper tests here. Unfortunately, I've just proposed some environment for the multiple configurations, not the functionality itself, so I am not ready to have responsibility for the functional part.